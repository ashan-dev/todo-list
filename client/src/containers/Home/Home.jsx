import React, { useState, useEffect } from "react";
import { Container, Row, Col } from "react-bootstrap";
import AddToDo from "../../components/AddToDo/AddToDo.jsx";
import ToDoList from "../../components/ToDoList/ToDoList.jsx";
import { connect } from "react-redux";
import {
  toDoDataListAction,
  updateToDoDatalistAction,
  deleteToDoDatalistAction,
} from "../../redux/actions/toDoList-action";


import styles from "./Home.module.scss";

const Home = ({
  toDoDataListProps,
  updatingToDoDatalistProps,
  todoList,
  deleteToDoDatalistProps,
}) => {
  const [task, setTask] = useState(null);

  const addingTaskHandler = () => {
    toDoDataListProps(task);
    setTask(null);
  };

  const onChangeTaskHandler = (task) => {
    setTask({
      task: task,
      completed: false,
    });
  };

  const completedTaskHandler = (id) => {
    updatingToDoDatalistProps(id);
  };

  const deletedTaskHandler = (id) => {
    deleteToDoDatalistProps(id);
  };
  return (
    <>
      <Container className={styles.homeContainer} fluid>
        <Row>
          <AddToDo
            toDoDataListProps={toDoDataListProps}
            onChangeTaskHandler={onChangeTaskHandler}
            addingTaskHandler={addingTaskHandler}
            task={task}
          />
          <ToDoList
            todoList={todoList}
            completedTaskHandler={completedTaskHandler}
            deletedTaskHandler={deletedTaskHandler}
          />
        </Row>
      </Container>
    </>
  );
};

const mapStateToProps = (state) => ({
  todoList: state.toDoListState.taskList,
});

const mapDispatchToProps = (dispatch) => ({
  toDoDataListProps: (payload) => dispatch(toDoDataListAction(payload)),
  updatingToDoDatalistProps: (payload) =>
    dispatch(updateToDoDatalistAction(payload)),
  deleteToDoDatalistProps: (payload) =>
    dispatch(deleteToDoDatalistAction(payload)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Home);
