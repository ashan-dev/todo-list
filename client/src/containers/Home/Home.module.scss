.homeContainer {
  padding-top: 50px;
  
}
// .sidebarWrapper {
//   background-color: red;

// }
// .pageContentWrapper {
//   background-color: green;

// }
