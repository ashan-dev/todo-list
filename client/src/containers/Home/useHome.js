import { useState } from "react";

const useHome = ({
  toDoDataListProps,
  updatingToDoDatalistProps,
  deleteToDoDatalistProps,
}) => {
  const [task, setTask] = useState(null);

  const addingTaskHandler = () => {
    if (task) {
      toDoDataListProps(task);
    }

    setTask(null);
  };

  const onChangeTaskHandler = (task) => {
    setTask({
      task: task,
      completed: false,
    });
  };

  const completedTaskHandler = (id) => {
    updatingToDoDatalistProps(id);
  };

  const deletedTaskHandler = (id) => {
    deleteToDoDatalistProps(id);
  };

  return [
    task,
    addingTaskHandler,
    onChangeTaskHandler,
    completedTaskHandler,
    deletedTaskHandler,
  ];
};

export default useHome;
