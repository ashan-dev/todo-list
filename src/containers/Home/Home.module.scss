.homeContainer {
  padding-top: 50px;
}
