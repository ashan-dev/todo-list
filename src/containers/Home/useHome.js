import { useState } from "react";

const useHome = ({
  toDoDataListProps,
  updatingToDoDatalistProps,
  deleteToDoDatalistProps,
}) => {
  const [task, setTask] = useState(null);
  const [editedTask, setEditedTask] = useState(null);

  const addingTaskHandler = () => {
    if (task) {
      toDoDataListProps(task);
    }
    setTask(null);
  };

  const onChangeTaskHandler = (task) => {
    setTask({
      task: task,
      completed: false,
    });
  };

  const onEditTaskHandler = (task) => {
    setEditedTask(task);
  };

  const editTaskHandler = (id) => {
    alert("test");
    console.log(id);
  };

  const completedTaskHandler = (id) => {
    updatingToDoDatalistProps(id);
  };

  const deletedTaskHandler = (id) => {
    deleteToDoDatalistProps(id);
  };

  return [
    {
      task,
      editedTask,
      addingTaskHandler,
      onChangeTaskHandler,
      onEditTaskHandler,
      completedTaskHandler,
      deletedTaskHandler,
      editTaskHandler,
    },
  ];
};

export default useHome;
