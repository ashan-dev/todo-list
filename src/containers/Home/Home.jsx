import React from "react";
import { Container, Row } from "react-bootstrap";
import AddToDo from "../../components/AddToDo/AddToDo.jsx";
import ToDoList from "../../components/ToDoList/ToDoList.jsx";
import useHome from "./useHome";
import { connect } from "react-redux";
import {
  toDoDataListAction,
  updateToDoDatalistAction,
  deleteToDoDatalistAction,
} from "../../redux/actions/toDoList-action";

import styles from "./Home.module.scss";

const Home = ({
  toDoDataListProps,
  updatingToDoDatalistProps,
  todoList,
  deleteToDoDatalistProps,
}) => {
  const [
    {
      task,
      editedTask,
      addingTaskHandler,
      onChangeTaskHandler,
      editTaskHandler,
      onEditTaskHandler,
      completedTaskHandler,
      deletedTaskHandler,
    },
  ] = useHome({
    toDoDataListProps,
    updatingToDoDatalistProps,
    deleteToDoDatalistProps,
  });

  return (
    <>
      <Container className={styles.homeContainer} fluid>
        <Row>
          <AddToDo
            toDoDataListProps={toDoDataListProps}
            onChangeTaskHandler={onChangeTaskHandler}
            addingTaskHandler={addingTaskHandler}
            task={task}
          />
          <ToDoList
            todoList={todoList}
            editedTask={editedTask}
            editTaskHandler={editTaskHandler}
            onEditTaskHandler={onEditTaskHandler}
            completedTaskHandler={completedTaskHandler}
            deletedTaskHandler={deletedTaskHandler}
          />
        </Row>
      </Container>
    </>
  );
};

const mapStateToProps = (state) => ({
  todoList: state.toDoListState?.taskList,
});

const mapDispatchToProps = (dispatch) => ({
  toDoDataListProps: (payload) => dispatch(toDoDataListAction(payload)),
  updatingToDoDatalistProps: (payload) =>
    dispatch(updateToDoDatalistAction(payload)),
  deleteToDoDatalistProps: (payload) =>
    dispatch(deleteToDoDatalistAction(payload)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Home);
