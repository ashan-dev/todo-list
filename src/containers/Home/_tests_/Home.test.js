import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Home from '../Home.jsx';

const mockStore = configureMockStore();
const store = mockStore({
  todoList: [],
});

test('Home renders correctly', () => {
  const component = renderer.create(
    <Provider store={store}>
      <Home />
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
