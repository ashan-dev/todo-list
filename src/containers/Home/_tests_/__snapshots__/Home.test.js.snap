// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div
  className="homeContainer container-fluid"
>
  <div
    className="row"
  >
    <div
      className="mb-3 row"
    >
      <div
        className="addTaskContainer col-10 offset-1"
      >
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Add a task"
          type="text"
          value=""
        />
        <div
          className="buttonContainer"
        >
          <svg
            className="button"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "#8fd5ff",
                "fontSize": 26,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M16 8A8 8 0 110 8a8 8 0 0116 0zM8.5 4a.5.5 0 00-1 0v3.5H4a.5.5 0 000 1h3.5V12a.5.5 0 001 0V8.5H12a.5.5 0 000-1H8.5V4z"
              fillRule="evenodd"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-10 offset-1"
      />
    </div>
  </div>
</div>
`;
